### PR TITLE
Generate ToC for README

### PR DESCRIPTION
**Pull request by paracycle**
### Motivation

Supersedes #1049 and #1068

### Implementation

Use the `kramdown` gem and our existing README processing script to generate a ToC for our README that is checked to stay in sync with its contents.

### Tests

No tests

### CODE REVIEW

File: tasks/readme.rake [hunk 1]
```diff
@@ -148,12 +148,35 @@ task :readme do
...
+
+      ["#{padding}- [#{text}](##{id})"] + convert_to_list_items(elem.children)
+    end
+  end
+
+  def print_toc(contents)
+    require "kramdown"
+    section = "TOC"
+
+    doc = Kramdown::Document.new(contents, auto_id: true, toc_levels: (2..6))
```
Comment (Morriar) [inline tasks/readme.rake:166]: Is it equivalent to using https://kramdown.gettalong.org/rdoc/Kramdown/Converter/Toc.html?

Comment (paracycle) [reply tasks/readme.rake:166]: The next line is equivalent to that. What you've linked to is a converter which works on a `Document` instance, which is this.

Comment (Morriar) [reply tasks/readme.rake:166]: Ahh `to_toc`, thanks!

File: tasks/readme.rake [hunk 2]
```diff
@@ -148,12 +148,35 @@ task :readme do
     contents
   end
 
+  def convert_to_list_items(elements)
+    elements.flat_map do |elem|
+      level = elem.value.options[:level]
+      text = elem.value.options[:raw_text]
```
[the remark this answers is not in the document]
Comment (paracycle) [reply tasks/readme.rake:154]: It just places the raw header text into the link as `[Some heading with a **bold** text](auto-generated-id)`.

**Review by Morriar — APPROVED** [review]

**Review by vinistock — APPROVED** [review]

**Pull request merged**